aptos-move/framework/aptos-framework/sources/nonce_validation.move

```diff
@@ -0,0 +1,148 @@
+module aptos_framework::nonce_validation {
+    use aptos_std::table::{Self, Table};
+    use aptos_std::timestamp;
+    use aptos_std::big_ordered_map::{Self, BigOrderedMap};
+    use aptos_std::aptos_hash::sip_hash_from_value;
+    use aptos_std::option;
+    use aptos_std::vector;
+    friend aptos_framework::genesis;
+    friend aptos_framework::transaction_validation;
+
+    const NUM_BUCKETS: u64 = 50000;
+    const MAP_SWITCH_TIME: u64 = 75;
+    const MAX_EXPIRATION_TIME: u64 = 60;
+
+    struct Bucket has store {
+        // Maps NonceKey = (address, nonce) to txn expiration time.
+        // Using multiple maps to facilitate easier garbage collection.
+        nonces: vector<BigOrderedMap<NonceKey, u64>>,
+        // last_stored_times[i] stores the timestamp at which a nonce entry 
+        // is last added to the nonces[i] map.
+        last_stored_times: vector<u64>,
+    }
+
+    struct NonceKey has copy, drop, store {
+        sender_address: address,
+        nonce: u64,
```

Comment:
why u64 as well?

Reply:
It's u64 in rust transaction type as well. Had to use the same datatype here.